door-open-service/esp8266-nodemcu/init.lua

```diff
@@ -13,16 +13,18 @@ gpio.write(gpio_pin, gpio.LOW)
 
 -- 3. Set-up timer
 function setup_alarm()
-  tmr.alarm(0, 3000, 1, function ()
-    local ip = wifi.sta.getip()
-    if ip then
-      tmr.stop(0)
+  if wifi.sta.getip() then
+    connect_to_sf_service()
+  else
+    configure_wifi()
```

Comment:
Maybe this should be after we hookup the `GOTIP` event and start monitoring, otherwise it looks like a possible race condition: what if it gets configured and get's an IP address before the `eventMonStart` takes place?